sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/MethodsShouldNotHaveTooManyLines_CustomValues.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public Program() // Noncompliant {{This constructor 'Program' has 4 lines, which is greater than the 2 lines authorized. Split it into smaller methods.}}
+//             ^^^^^^^
+        {
+            int i = 1; i++;
+            i++;
+            i++;
+            i++;
+        }
+
+        public ~Program() // Noncompliant {{This finalizer '~Program' has 4 lines, which is greater than the 2 lines authorized. Split it into smaller methods.}}
+        {
+            int i = 1; i++;
+            i++;
+            i++;
+            i++;
+        }
+
+
+        public void Method_01() // Noncompliant {{This method 'Method_01' has 4 lines, which is greater than the 2 lines authorized. Split it into smaller methods.}}
+//                  ^^^^^^^^^
+        {
+            int i = 1; i++;
+            i++;
+            i++;
+            i++;
+        }
+
+        public int Method_02() // Noncompliant {{This method 'Method_02' has 13 lines, which is greater than the 2 lines authorized. Split it into smaller methods.}}
+        {
+            int i = 1;
+            i++;
+            i++;
+            i++;
+            i++;
+            i++;
+            i++;
+            i++;
+            i++;
+            i++;
+            i++;
+            i++;
+
+            return 1;
+        }
+
+        public abstract int Method_04();
+
+        extern int Method_05();
+
+        public int Method_06() // Noncompliant {{This method 'Method_06' has 6 lines, which is greater than the 2 lines authorized. Split it into smaller methods.}}
+        {
+
+            void InnerFunction_06()
+            {
+                int i = 0;
+                i++;
+                i++;
+            }
+        }
+
+        // Compliant. Properties are not checked
```

Comment:
Properties are like methods imo and they should be checked...

Reply:
Good point, was wondering about it myself (see how good I am with asking questions indirectly using code? :) ). I've discussed it with Ann, she suggested to create a separate rule for that. I'll discuss it on dogfood.